modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -990,6 +1088,37 @@ void ONNXImporter::populateNet(Net dstNet)
                 continue;
             }
         }
+        else if (layer_type == "Resize")
+        {
+            for (int i = 1; i < node_proto.input_size(); i++)
+                CV_Assert(layer_id.find(node_proto.input(i)) == layer_id.end());
+
+            String interp_mode = layerParams.get<String>("coordinate_transformation_mode");
+            CV_Assert_N(interp_mode != "tf_crop_and_resize", interp_mode != "asymmetric",
+                        interp_mode != "tf_half_pixel_for_nn");
+
+            layerParams.set("align_corners", interp_mode == "align_corners");
+            Mat shapes = getBlob(node_proto, constBlobs, node_proto.input_size() - 1);
+            CV_Assert_N(shapes.size[0] == 4, shapes.size[1] == 1);
```

Comment:
Please add type check,
```
CV_Assert_N(shapes.size[0] == 4, shapes.size[1] == 1, shapes.type == CV_32S);
```

Reply:
> CV_Assert_N

Just try to replace `CV_32S` => `CV_64F` to check error message.
`CV_CheckTypeEQ()` should be used instead.